From 8c1f7a7ac3c897b03a8f2a2b0f58e63c2e249ff3 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Thu, 9 Mar 2017 18:03:13 +0200
Subject: [PATCH 0860/1345] dma: mv_xor_v2: remove Armada-8k-Z1 support &
 update filed name

commit  fc17a758920c58ff97aec21c988b60d88270d6ae from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

No functional changes in this patch.
Remove Armada-8k-Z1 support from XOR v2 dirver.
Split the src field in HW structure to be src & dst.

Change-Id: I4db9ab40bfc5c71c513e3d61cacb8358d64fa953
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/37412
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/mv_xor_v2.c |   42 +++++-------------------------------------
 1 file changed, 5 insertions(+), 37 deletions(-)

diff --git a/drivers/dma/mv_xor_v2.c b/drivers/dma/mv_xor_v2.c
index 0ae7b0d..1815c37 100644
--- a/drivers/dma/mv_xor_v2.c
+++ b/drivers/dma/mv_xor_v2.c
@@ -68,10 +68,8 @@
 #define   MV_XOR_V2_GLOB_BW_CTRL_NUM_OSTD_WR_VAL	8
 #define   MV_XOR_V2_GLOB_BW_CTRL_RD_BURST_LEN_SHIFT	12
 #define   MV_XOR_V2_GLOB_BW_CTRL_RD_BURST_LEN_VAL	4
-#define   MV_XOR_V2_GLOB_BW_CTRL_RD_BURST_LEN_VAL_Z1	2
 #define   MV_XOR_V2_GLOB_BW_CTRL_WR_BURST_LEN_SHIFT	16
 #define	  MV_XOR_V2_GLOB_BW_CTRL_WR_BURST_LEN_VAL	4
-#define   MV_XOR_V2_GLOB_GLOB_BW_CTRL_WR_BURST_LEN_VAL_Z1     	2
 #define MV_XOR_V2_GLOB_PAUSE				0x014
 #define   MV_XOR_V2_GLOB_PAUSE_AXI_TIME_DIS_VAL		0x8
 #define MV_XOR_V2_GLOB_SYS_INT_CAUSE			0x200
@@ -192,17 +190,6 @@ struct mv_xor_v2_sw_desc {
 };
 
 /*
- * Hold XOR engine parameters depending on the matched
- * compatible string.
- */
-struct mv_xor_v2_compat_data {
-	u32 rd_burst_len;
-	u32 wr_burst_len;
-};
-
-static const struct of_device_id mv_xor_v2_dt_ids[];
-
-/*
  * Fill the data buffers to a HW descriptor
  */
 static void mv_xor_v2_set_data_buffers(struct mv_xor_v2_device *xor_dev,
@@ -649,8 +636,6 @@ static void mv_xor_v2_set_msi_msg(struct msi_desc *desc, struct msi_msg *msg)
 static int mv_xor_v2_descq_init(struct mv_xor_v2_device *xor_dev)
 {
 	u32 reg;
-	const struct of_device_id *match;
-	struct mv_xor_v2_compat_data *data;
 
 	/* write the DESQ size to the DMA engine */
 	writel(MV_XOR_V2_DESC_NUM,
@@ -690,17 +675,14 @@ static int mv_xor_v2_descq_init(struct mv_xor_v2_device *xor_dev)
 	 * -  Limit the number of outstanding write & read data
 	 *    (OBB/IBB) requests to the maximal value.
 	*/
-	match = of_match_node(mv_xor_v2_dt_ids, xor_dev->dmadev.dev->of_node);
-	if (WARN_ON(!match))
-		return -1;
-	data = (struct mv_xor_v2_compat_data *)match->data;
-
 	reg = ((MV_XOR_V2_GLOB_BW_CTRL_NUM_OSTD_RD_VAL <<
 		MV_XOR_V2_GLOB_BW_CTRL_NUM_OSTD_RD_SHIFT) |
 	       (MV_XOR_V2_GLOB_BW_CTRL_NUM_OSTD_WR_VAL  <<
 		MV_XOR_V2_GLOB_BW_CTRL_NUM_OSTD_WR_SHIFT) |
-		(data->rd_burst_len << MV_XOR_V2_GLOB_BW_CTRL_RD_BURST_LEN_SHIFT) |
-		(data->wr_burst_len << MV_XOR_V2_GLOB_BW_CTRL_WR_BURST_LEN_SHIFT);
+	       (MV_XOR_V2_GLOB_BW_CTRL_RD_BURST_LEN_VAL <<
+		MV_XOR_V2_GLOB_BW_CTRL_RD_BURST_LEN_SHIFT) |
+	       (MV_XOR_V2_GLOB_BW_CTRL_WR_BURST_LEN_VAL <<
+		MV_XOR_V2_GLOB_BW_CTRL_WR_BURST_LEN_SHIFT));
 	writel(reg, xor_dev->glob_base + MV_XOR_V2_GLOB_BW_CTRL);
 
 	/* Disable the AXI timer feature */
@@ -907,22 +889,8 @@ static int mv_xor_v2_remove(struct platform_device *pdev)
 }
 
 #ifdef CONFIG_OF
-
-struct mv_xor_v2_compat_data xor_compat_data = {
-	.rd_burst_len = MV_XOR_V2_GLOB_BW_CTRL_RD_BURST_LEN_VAL,
-	.wr_burst_len = MV_XOR_V2_GLOB_BW_CTRL_WR_BURST_LEN_VAL
-};
-
-struct mv_xor_v2_compat_data xor_compat_data_z1 = {
-	.rd_burst_len = MV_XOR_V2_GLOB_BW_CTRL_RD_BURST_LEN_VAL_Z1,
-	.wr_burst_len = MV_XOR_V2_GLOB_GLOB_BW_CTRL_WR_BURST_LEN_VAL_Z1
-};
-
 static const struct of_device_id mv_xor_v2_dt_ids[] = {
-	{ .compatible = "marvell,mv-xor-v2",
-	  .data = &xor_compat_data},
-	{ .compatible = "marvell,mv-xor-v2-z1",
-	  .data = &xor_compat_data_z1 },
+	{ .compatible = "marvell,xor-v2", },
 	{},
 };
 
-- 
1.7.9.5

